openssl/src/ssl/mod.rs

```diff
@@ -1718,6 +1718,89 @@ impl SslContextBuilder {
         unsafe { cvt(ffi::SSL_CTX_set_num_tickets(self.as_ptr(), num_tickets)).map(|_| ()) }
     }
 
+    /// initialize the shared state required for DANE support.
+    ///
+    /// Requires OpenSSL 1.1.0 or newer.
+    #[corresponds(SSL_CTX_dane_enable)]
+    #[cfg(ossl110)]
+    pub fn dane_enable(&mut self) -> Result<(), ErrorStack> {
+        unsafe { cvt(ffi::SSL_CTX_dane_enable(self.as_ptr())).map(|_| ()) }
+    }
+
+    /// Adjust the supported digest algorithms.
+    /// This must be done before any SSL handles are created for the context.
+    ///
+    /// The mtype argument specifies a DANE TLSA matching type and the md argument
+    /// specifies the associated digest algorithm handle.
+    /// The ord argument specifies a strength ordinal.
+    ///
+    /// Algorithms with a larger strength ordinal are considered more secure.
+    ///
+    /// Strength ordinals are used to implement RFC7671 digest algorithm agility.
+    ///
+    /// Specifying None for the digest algorithm for a matching type disables
+    /// support for that matching type.
+    ///
+    /// Matching type Full(0) cannot be modified or disabled.
+    ///
+    /// By default, matching type SHA2-256(1) (see RFC7218 for definitions of
+    /// the DANE TLSA parameter acronyms) is mapped to EVP_sha256() with a strength
+    /// ordinal of 1 and matching type SHA2-512(2) is mapped to EVP_sha512()
+    /// with a strength ordinal of 2.
+    ///
+    /// Requires OpenSSL 1.1.0 or newer.
+    #[corresponds(SSL_CTX_dane_mtype_set)]
+    #[cfg(ossl110)]
+    pub fn dane_mtype_set(
+        &mut self,
+        md: Option<crate::md::MdRef>,
+        mtype: u8,
+        ord: u8,
```

Comment:
These should both be passed as more strongly typed values - see `Asn1Type` for example.